src/utils/instrumentation.ts

```diff
@@ -0,0 +1,201 @@
+          {
+            // Truncate libraryName and libraryVersion if more than 14 characters length
+            name: truncateValue(libraryName),
+            version: truncateValue(libraryVersion ?? getNodejsLibraryVersion()),
+          },
+        ],
+      },
+    }
+  );
+  return entry;
+}
+
+/**
+ * This method validates that provided instrumentation info list is valid and also adds current library info to a list.
+ * @param infoList {InstrumentationInfo} The array of InstrumentationInfo to be validated and updated.
+ * @returns {InstrumentationInfo} The updated list of InstrumentationInfo.
+ */
+function validateAndUpdateInstrumentation(
+  infoList: InstrumentationInfo[]
+): InstrumentationInfo[] {
+  const finalInfo: InstrumentationInfo[] = [];
+  // First, add current library information
+  finalInfo.push({
+    name: NODEJS_LIBRARY_NAME_PREFIX,
+    version: getNodejsLibraryVersion(),
+  });
+  // Iterate through given list of libraries and for each entry perform validations and transformations
+  // Limit amount of entries to be up to 3
+  let count = 1;
+  for (const info of infoList) {
+    if (isValidInfo(info)) {
+      finalInfo.push({
+        name: truncateValue(info.name),
+        version: truncateValue(info.version),
+      });
+    }
+    if (++count === 3) break;
+  }
+  return finalInfo;
+}
+
+/**
+ * A helper function to truncate a value (library name or version). The value is truncated
+ * when it is longet than 14 chars and '*' is added instead of truncated suffix.
+ * @param value {string} The library name or version to be truncated.
+ * @returns {string} The truncated value.
+ */
+function truncateValue(value: string) {
+  if (value && value.length > maxDiagnosticValueLen) {
+    return value.substring(0, maxDiagnosticValueLen).concat('*');
+  }
+  return value;
+}
+
+/**
+ * The helper function to retrieve current library version from 'package.json' file. Note that
+ * since we use {path.resolve}, the search for 'package.json' could be impacted by current working directory.
+ * @returns {string} A current library version.
+ */
+export function getNodejsLibraryVersion() {
+  if (libraryVersion) {
+    return libraryVersion;
+  }
+  libraryVersion = require(path.resolve(
+    __dirname,
+    '../../../',
+    'package.json'
+  )).version;
+  return libraryVersion;
+}
+
+/**
+ * The helper function which checks if given InstrumentationInfo is valid.
+ * @param info {InstrumentationInfo} The info to be validated.
+ * @returns true if given info is valid, false otherwise
+ */
+function isValidInfo(info: InstrumentationInfo) {
+  if (
+    !info ||
+    !info.name ||
+    !info.version ||
+    !info.name.startsWith(NODEJS_LIBRARY_NAME_PREFIX)
+  ) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * The helper method used for tests to reset a status if instrumentation already written.
+ */
+export function testResetInstrumentationStatus() {
+  global.instrumentationAdded = false;
+}
```

Comment:
nit: i hope there is a method to avoid placing test related code into the body of the exported package. maybe worth to consult with Benjamin

Reply:
I asked of course, but seems Nodejs do not have magic solutions here. I am thinking to make this method to be non-test related, WDYT?